Source/monster.cpp

```diff
@@ -5603,15 +5602,17 @@ void TalktoMonster(int i)
 	pnum = Monst->_menemy;
 	Monst->_mmode = MM_TALK;
 	if (Monst->_mAi == AI_SNOTSPIL || Monst->_mAi == AI_LACHDAN) {
-		if (QuestStatus(Q_LTBANNER) && quests[Q_LTBANNER]._qvar1 == 2 && PlrHasItem(pnum, IDI_BANNER, &itm)) {
-			RemoveInvItem(pnum, itm);
+		itm = plr[pnum].GetItemIndex(IDI_BANNER);

```

Comment:
I'd prefer to keep this in the condition so that it is short-circuited and not retrieved when not needed.

It obviously requires a change in interface. See comment above (perhaps rename it to `bool TryGetItemIndex(ref itemIndex)`)

Reply:
Maybe it would be and idea to make index optional as it's not always used. Or maybe we should really have two functions?

Reply:
I proposed a `bool TryGetItem(ItemStruct *)` in another comment. It is more general purpose, it requires no " != -1" checks, and you can change the logic that relies on indexes today (remove item from inventory by index) to rely on the item reference instead).

Reply:
im not sure how i would delete by reference 😅 set its type to empty and then scan the whole inventory for empty items?

Reply:
> im not sure how i would delete by reference

It's not 100% clear to me as well, but it seems more intuitive to go with something like that. I'd have to experiment a bit with it myself as well.

> set its type to empty and then scan the whole inventory for empty items?

Maybe... perhaps @glebm has a suggestion here?

Reply:
std::remove works with both iterators for C++ containers and pointers for C arrays

Reply:
problem here is setting the inventory grid to zero for where the item was located. doing a search for it probably isent the end of the world, but ill have to look at it